src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+            return sAnjuMsgScript_80B58B88;
+        }
+
+        if (this->scheduleResult == ANJU_SCH_MIDNIGHT_MEETING) {
+            this->msgEventFunc = EnAn_MsgEvent_MidnightMeeting;
+            return sAnjuMsgScript_SchMidnightMeeting;
+        }
+
+        if (this->scheduleResult == ANJU_SCH_LAUNDRY_POOL_SIT) {
+            this->msgEventFunc = EnAn_MsgEvent_LaundryPool;
+            return sAnjuMsgScript_SchLaundryPoolKafeiMask;
+        }
+
+        if (gSaveContext.save.day >= 2) {
+            return sAnjuMsgScript_80B58A44;
+        }
+
+        if (SCHEDULE_TIME_NOW < SCHEDULE_TIME(14, 18)) {
+            return sAnjuMsgScript_80B58ABC;
+        }
+
+        return sAnjuMsgScript_80B58AC4;
+    }
+
+    if ((player->transformation == PLAYER_FORM_HUMAN) && CHECK_WEEKEVENTREG(WEEKEVENTREG_PROMISED_MIDNIGHT_MEETING)) {
+        if (this->scheduleResult == ANJU_SCH_MIDNIGHT_MEETING) {
+            this->msgEventFunc = EnAn_MsgEvent_MidnightMeeting;
+            return sAnjuMsgScript_SchMidnightMeeting;
+        }
+
+        if (this->scheduleResult == ANJU_SCH_LAUNDRY_POOL_SIT) {
+            this->msgEventFunc = EnAn_MsgEvent_LaundryPool;
+            return sAnjuMsgScript_SchLaundryPoolDefault;
+        }
+
+        if (gSaveContext.save.day >= 2) {
+            return sAnjuMsgScript_80B58A44;
+        }
+    }
+
+    switch (this->scheduleResult) {
+        case ANJU_SCH_WAITING_FOR_KAFEI:
+            return sAnjuMsgScript_SchWaitingForKafei;
+
+        case ANJU_SCH_WALKING_40:
+            return sAnjuMsgScript_80B5894C;
+
+        case ANJU_SCH_WALKING_42:
+        case ANJU_SCH_WALKING_43:
+        case ANJU_SCH_WALKING_44:
+        case ANJU_SCH_WALKING_45:
+            return sAnjuMsgScript_80B58980;
+
+        case ANJU_SCH_LAUNDRY_POOL_SIT:
+            this->msgEventFunc = EnAn_MsgEvent_LaundryPool;
+            return sAnjuMsgScript_SchLaundryPoolDefault;
+
+        case ANJU_SCH_RECEPTIONIST_IDLE:
+            if (gSaveContext.save.day >= 2) {
+                return sAnjuMsgScript_Receptionist;
+            }
+            return sAnjuMsgScript_ReceptionistDay1;
+
+        case ANJU_SCH_WAITING_CLOSING_TIME:
+        case ANJU_SCH_WALKING_47:
+            return sAnjuMsgScript_InnCloseTime;
+
+        case ANJU_SCH_SWEEPING:
+            return sAnjuMsgScript_SchSweeping;
+
+        case ANJU_SCH_MIDNIGHT_MEETING:
+            return sAnjuMsgScript_80B589FC;
+
+        case ANJU_SCH_WALKING_48:
+            return sAnjuMsgScript_80B58944;
+
+        case ANJU_SCH_WALKING_49:
+            if (player->transformation == PLAYER_FORM_HUMAN) {
+                return sAnjuMsgScript_80B58B88;
+            }
+            return sAnjuMsgScript_80B589FC;
+
+        case ANJU_SCH_WALKING_52:
+        case ANJU_SCH_WALKING_53:
+            return sAnjuMsgScript_80B58994;
+
+        default:
+            break;
+    }
+
+    return NULL;
+}
+
+s32 EnAn_CheckTalk(EnAn* this, PlayState* play) {
```

Comment:
Maybe `EnAn_TryTalking`?

Reply:
I used this name because it is similar to the name of other schedule actors (`EnBjt_CheckTalk` and `EnPst_CheckTalk`)
https://github.com/zeldaret/mm/blob/28e2da2a18ddf1fc2dd81199e73eb18252684b43/src/overlays/actors/ovl_En_Bjt/z_en_bjt.c#L320-L333
https://github.com/zeldaret/mm/blob/28e2da2a18ddf1fc2dd81199e73eb18252684b43/src/overlays/actors/ovl_En_Pst/z_en_pst.c#L279-L304

Reply:
Hmm I see. I like `TryTalking` better for both of them, but you can leave it as `CheckTalk` for consistency for now if you want

Reply:
We should do a cleanup pass for all the schedule actors in the future. Specially now that we have all of them matching.
The only thing that could be worth waiting for is the msgevent system tho.
So yeah, I'll keep `CheckTalk` for consistency